packages/cli-hydrogen/src/cli/services/deploy/deployer.ts

```diff
@@ -92,6 +92,22 @@ export const uploadDeploymentStep = async (config: ReqDeployConfig, deploymentID
   return responseData.data.uploadDeployment.deployment.previewURL
 }
 
+export const healthCheck = async (url: string): Promise<boolean> => {
+  output.info('📡 Checking deployment... ')
+  const pingUrl = `${url}/__health`
+  const rollBack = [5, 10, 15, 30, 60]
+
+  for (const seconds of rollBack) {
+    /* eslint-disable no-await-in-loop */
+    const result = await http.fetch(pingUrl, {method: 'GET'})
+    if (result.status === 200) return true
+    await system.sleep(seconds)
+    /* eslint-enable no-await-in-loop */
+  }
+
+  return false
+}
+
```

Comment:
Do we want to extract this "retry" logic into a separate method that we can for our other methods? As part of the "refactor" I think all of our fetch / graphql calls are going to need to be able to retry so might be worth making it generalized.